### PR TITLE
Add options to customize leader election

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:
Add options to customize leader election

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by shyamradhakrishnan** [comment]
e2e test
```
------------------------------

Ran 6 of 25 Specs in 4025.928 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 19 Skippe
```

File: main.go [hunk 1]
```diff
@@ -18,7 +18,9 @@ package main
 
 import (
 	"flag"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
```
Comment (joekr) [inline main.go:21]: can you move this down in to the external dependency list

**Review by joekr — DISMISSED** [review]
Small NIT

File: main.go (hunk 1 above)
Comment (shyamradhakrishnan) [reply main.go]: fixed..

**Review by joekr — APPROVED** [review]

**Pull request merged**